dotcom-rendering/src/web/components/HeaderSingleFrontDoor.tsx

```diff
@@ -31,37 +32,43 @@ export const HeaderSingleFrontDoor = ({
 	remoteHeader,
 	contributionsServiceUrl,
 	idApiUrl,
-}: Props) => (
-	<div css={headerStyles}>
-		<Island deferUntil="idle">
-			<HeaderTopBar
-				editionId={editionId}
-				dataLinkName="nav3 : topbar : edition-picker: toggle"
-				idUrl={idUrl}
-				mmaUrl={mmaUrl}
-				discussionApiUrl={discussionApiUrl}
-				idApiUrl={idApiUrl}
-			/>
-		</Island>
-		<div
-			css={[
-				center,
-				css`
-					overflow: auto;
-				`,
-			]}
-		>
-			<Logo editionId={editionId} />
-			<Island deferUntil="idle" clientOnly={true}>
-				<SupportTheG
-					urls={urls}
+}: Props) => {
+	const isServer = typeof window === 'undefined';
+	const isSignedIn =
+		!isServer && !!getCookie({ name: 'GU_U', shouldMemoize: true });
```

Comment:
This can live outside of the component and does not have to be regenerated on each render.